src/main/java/org/jabref/logic/citation/repository/MVStoreBibEntryRelationDAO.java

```diff
@@ -0,0 +1,238 @@
+package org.jabref.logic.citation.repository;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.time.Clock;
+import java.time.ZoneId;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.time.LocalDateTime;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.identifier.DOI;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.h2.mvstore.MVMap;
+import org.h2.mvstore.MVStore;
+import org.h2.mvstore.WriteBuffer;
+import org.h2.mvstore.type.BasicDataType;
+
+public class MVStoreBibEntryRelationDAO implements BibEntryRelationDAO {
+
+    private final static ZoneId TIME_STAMP_ZONE_ID = ZoneId.of("UTC");
+
+    private final String mapName;
+    private final String insertionTimeStampMapName;
+    private final MVStore.Builder storeConfiguration;
+    private final MVMap.Builder<String, LinkedHashSet<BibEntry>> mapConfiguration =
+        new MVMap.Builder<String, LinkedHashSet<BibEntry>>().valueType(new BibEntryHashSetSerializer());
+
+    MVStoreBibEntryRelationDAO(Path path, String mapName) {
+        this.mapName = mapName;
+        this.insertionTimeStampMapName = mapName + "-insertion-timestamp";
+        this.storeConfiguration = new MVStore.Builder().autoCommitDisabled().fileName(path.toAbsolutePath().toString());
```

Comment:
It would be nice if you place such chained methods in several lines, like this:

```java
this.storeConfiguration = new MVStore.Builder()
	.autoCommitDisabled()
	.fileName(path.toAbsolutePath().toString());
```

That is what we often use.